StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/Lightup/SwitchExpressionArmSyntaxWrapperTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Test.CSharp8.Lightup
+{
+    using System;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Lightup;
+    using Xunit;
+
+    public class SwitchExpressionArmSyntaxWrapperTests
```

Comment:
💡 This also needs tests for C# 6 and C# 7. Since these two will behave the same way, you can implement this test once (in the C# 6 tests), and then create an empty derived type in the C# 7 test project:

```csharp
public class CSharp7SwitchExpressionArmSyntaxWrapperTests : SwitchExpressionArmSyntaxWrapperTests
{
}
```

Reply:
I think I cannot move all the tests on the C# 6 project because the `SyntaxFactory.ConstantPattern` and `SyntaxFactory.WhenClause` are available only starting with C# 7 and `SyntaxFactory.SwitchExpressionArm` with C# 8.

What I did in the end is to split the tests according with their availability.

Reply:
Here's an example of split tests:

* C# 6 (syntax not available): https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/c69ba488c8d23e23bf76772073ccb5cbacc7e319/StyleCop.Analyzers/StyleCop.Analyzers.Test/Lightup/DeclarationPatternSyntaxWrapperTests.cs
* C# 7 (syntax available): https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/c69ba488c8d23e23bf76772073ccb5cbacc7e319/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp7/Lightup/DeclarationPatternSyntaxWrapperTests.cs

Reply:
Ah, you should be good with the split approach here. It works for cases where the syntax is completely absent in the earlier releases, but not for cases where the syntax was just *changed*.

Reply:
Yes, I think the approach I've used is similar. I've tried to avoid duplication where possible,